File: frontend/src/api/admin/auditLogs.ts
import request from '../request'
import type { ApiResponse } from '../auth'

export interface AuditLogItem {
  log_id: number
  operator_id: number
  operator_name: string
  action: string
  resource_type: string
  resource_id: number | null
  detail: any
  ip_address: string | null
  user_agent: string | null
  created_at: string | null
}

export interface AuditLogListData {
  items: AuditLogItem[]
  total: number
  page: number
  page_size: number
}

export function getAuditLogs(params: {
  page?: number
  page_size?: number
  action?: string
  operator_name?: string
  resource_type?: string
  start_time?: string
  end_time?: string
}): Promise<ApiResponse<AuditLogListData>> {
  return request.get('/admin/audit-logs', { params })
}

File: backend/app/api/admin/audit_logs.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession
from app.core.database import get_db
from app.core.deps import require_roles
from app.models.user import User
from app.services.audit_service import AuditService
from app.utils.response import success

router = APIRouter(prefix="/admin/audit-logs", tags=["审计日志"], dependencies=[Depends(require_roles(["super_admin"]))])


@router.get("")
async def list_logs(
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
    action: str = None,
    operator_name: str = None,
    resource_type: str = None,
    start_time: str = None,
    end_time: str = None,
    db: AsyncSession = Depends(get_db),
):
    """审计日志列表"""
    service = AuditService(db)
    logs, total = await service.list_logs(
        page=page, page_size=page_size,
        action=action, operator_name=operator_name,
        resource_type=resource_type,
        start_time=start_time, end_time=end_time,
    )
    items = [
        {
            "log_id": log.log_id,
            "operator_id": log.operator_id,
            "operator_name": log.operator_name,
            "action": log.action,
            "resource_type": log.resource_type,
            "resource_id": log.resource_id,
            "detail": log.detail,
            "ip_address": log.ip_address,
            "user_agent": log.user_agent,
            "created_at": log.created_at.isoformat() if log.created_at else None,
        }
        for log in logs
    ]
    return success(data={"items": items, "total": total, "page": page, "page_size": page_size})